subprojects/core-api/src/main/java/org/gradle/api/artifacts/ComponentSelection.java

```diff
@@ -32,8 +35,34 @@
      *
      * @return the candidate version of the module
      */
+    @Nonnull
```

Comment:
:x: `@Nonnull` should be set by default on the package level using `@NonNullApi` api instead

Reply:
And that's already done, so removing the annotation here.